### DISCUSSION TITLE
Please make the app easier to use.

### DISCUSSION THREAD

**Discussion by d4f5409d** [osmandapp/OsmAnd#12343] [General]
Less functions/options, better user experience.

**Comment by vshcherb** [comment]
Could you specify what do you mean by that.

1. What functions didn't work as you expected? 
2. How do you think removing one functions could improve another functions? If you have a concrete example, that could help to understand.

**Reply by barteis** [reply]
There are many GPS apps with fewer functions and an easier learning curve.  Conversely, few GPS apps function at OsmAnd's level.  OsmAnd requires more effort, for sure.  And offers more rewards.

**Reply by d4f5409d** [reply]
For example, there shouldn't be different driving profiles, there should be only one so it would be less confusing. Or more profiles than 1 should be optional.

**Reply by scaidermern** [reply]
Different profiles are essential because different modes of transportation exist.

**Reply by vshcherb** [reply]
Indeed different profiles is the only way to customize app for different usage in case you would like to be a cyclist and a pedestrian at different moment of time to say. All navigation apps have different modes but in most of them you can't enter into different profile unless you start navigation.

**Reply by Lee-Carre** [reply]
> more profiles than 1 should be optional

That's already the case; disable the ones you won't use.

**Comment by Lee-Carre** [comment]
> Less functions/options, better user experience.

As someone moderately-well versed in usability & interaction-design, I thought I'd give some insight.

Firstly, your OP is about as vague (& subjective; define ‘better’) as could be. So much so as to be mostly useless for inspiring change (let alone improvement).

To avoid a long read for someone who may not appreciate it: if you want (the appearance of) ‘simplicity’, then perhaps Organic Maps is better suited for you.
Just don't assume that because OsmAnd isn't to your immediate liking, that it's somehow bad or been developed carelessly.

Other primer context includes (but is certainly not limited to):
* usability is nothing to do with ‘intuition’
* a lot goes into something being well-designed, this is no less true for usability
* different user-groups exist; different folks expect different things, have different personality-types, skill-level (read: tolerance for complexity), different domain-specific knowledge and (general) technical knowledge / experience

Secondly, the assumption that there is a trade-off between features & (for the simple use-case) usability, is largely false (the exception tends to be at the extremes).

[Edited to add:] Arguably, they correlate: [Complexity is Good, Simplicity Overrated](//jnd.org/complexity_is_good_simplicity_overrated_misc_magazine/). Automating away the mundane & repetitive involves *adding* complexity (to the code) in order to yield a better experience for the user.

Even if that weren't the case, which features to cut? Oh, the ones you don't like / use? Well, others like & use them.

There's no need to *remove* features (just ‘hide’ them until needed). Instead, the solution is to not burden users who only want to do simple & obvious tasks (Joe wants directions for his morning commute) with a UI which tries to show off every feature (not that I'm saying OsmAnd does this).

A good (well-designed, well-implemented, effective) UI is one which presents the right information at the right time (or other relevant conditions / circumstances). The details of ‘right info’, ‘right time’ (and, by extension, in the right way) are where it becomes tricky, quickly. Guesswork (as opposed to proper user-testing) abound. Getting it right is what yields a good UX.

In technical terms, for a moment, more features, and refinement of those features, is what makes software more capable. Imagine a ‘lite’ version which omits a feature you consider vital; it's useless to you, then. Yet, this is the case for every feature.
I, personally, have little need of most GPX features, but I recognise that many users rely on them. So, I just ignore most everything GPX related. I'm pleased that they're there for those who do need & want them, and for when I might want them in future.

This is why quality software often has a modular plugin system; so that each user can have only the features (beyond the essential core) that he wants. Different for each user.

Quality software rewards curiosity (exploring features) without penalising the obvious simple use-cases.

Features are not always user-visible, either. For example, I very much like that OsmAnd avoids many of the routing pitfalls of other implementations. I've seen some do very daft things, indeed. OsmAnd handles a lot of things gracefully, without fuss. That's largely down to algorithms and careful implementation (e.g. making some behaviours conditional). All due to the very complexity which the ill-informed complain about.

Earlier, profiles were criticised. I'll use that as a specific example. Imagine the inverse; no profiles (or an inadequate selection of presets). Well, then you've got to re-tweak parameters each time (between car and truck and motorbike, between bus and train, between hiking in the wilderness and being a pedestrian in an urban environment). So, there's a strong argument that profiles significantly *increase* usability. Once configured, they make navigation (by which I mean reconfiguration of various parameters, depending on context) simpler.
More profiles than you need? Well, ignore or disable the ones irrelevant to you.

Not only are there significant differences between a car and a truck (having plenty experience of both), but since OsmAnd helpfully allows inputting a vehicle's parameters (width, height, weight, length), then these may be different between a small car versus a large one. In which case, 2 different car profiles might make sense for some, to set-and-forget the details once, and have them abstracted away.

Imagine that wasn't possible, and you end up being routed down a narrow road in a wide vehicle.

It seems like many users of OsmAnd like having granular control, and refining the configuration in order to improve behaviour (as each of them define ‘improve’ and see fit to modify parameters; empowering each of them). That effort is rewarded by an increase of it-just-works automation and (dare I say) simplicity of day-to-day use.

Compare setting your heating & ventilation system just right, so that you're always comfortable while keeping costs low. As opposed to having to manually fiddle with it all the time.
I'd strongly argue that the former is a better system; because you can always ignore its capabilities and keep it in manual-mode to fiddle with the set-point to your heart's content.

Also keep in mind that, in the grand scheme, the whole OSM realm is still in its infancy. Early on, things tend to be more facilitating of tinkering & refinement, while the details are resolved. Not only is the map data incomplete, but routing algorithms are still very much an open-question field, as are data formats, and very likely other areas.
There are pros and cons of being an early-adopter.

If you really wanna help, then arrange some proper user-testing, and publish the findings for the developers to read.

Else, make much more specific requests (of people who are donating their time to develop libre-software).
